src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1891,7 +1891,7 @@ private BoundExpression SynthesizeReceiver(SyntaxNode node, Symbol member, Diagn
                 (currentType.IsInterface && (declaringType.IsObjectType() || currentType.AllInterfacesNoUseSiteDiagnostics.Contains(declaringType))))
             {
                 bool hasErrors = false;
-                if (EnclosingNameofArgument != node)
```

Comment:
@gafter I think you wrote this code originally. Was it intentional to not allow `nameof(InstanceProperty.Member)` in a static context, but allow `nameof(InstanceProperty)`?